opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/MessagesContainer.java

```diff
@@ -246,11 +264,21 @@ public boolean isExpired() {
             return getExpirationTime().isBefore(Instant.now());
         }
 
+        @JsonProperty("expiration")
         @JsonSerialize(using = InstantSerializer.class)
         public Instant getExpirationTime() {
             return acceptedTime.plus(message.getDuration());
         }
 
+        public String toJSON() {
```

Comment:
maybe it'd be worth it to extract this to an interface Jsonable with default method implementation to
1. not to duplicate the code
2. mark classes as jsonable

Reply:
The `private static String taggedMessagesToJson(Set<TaggedMessagesContainer> messages)` was left as a copy of `toJSON()`. Ideas on how to make the `Set` `JSONable` are welcome.

Reply:
I guess only by introducing another named class (like taggedmessagecontainer) extending Set and implementing Jsonable

But I don't have a full context on the usage so maybe one helper function is the way to go .